### PR TITLE
predicates/traffic: deflake TestTrafficSegmentSplit

### LINKED ISSUE #2665: Flaky TestTrafficSegmentSplit
```
--- FAIL: TestTrafficSegmentSplit (1.49s)
    segment_test.go:164: Response codes: map[200:5074 201:3085 202:1841]
    segment_test.go:168: 
        	Error Trace:	/home/runner/work/skipper/skipper/predicates/traffic/segment_test.go:93
        	            				/home/runner/work/skipper/skipper/predicates/traffic/segment_test.go:168
        	Error:      	Max difference between 2000 and 1841 allowed is 120, but difference was 159
        	Test:       	TestTrafficSegmentSplit
```

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
Use fixed random sequence to deflake TestTrafficSegmentSplit.

Fixes #2665

### CODE REVIEW

File: predicates/traffic/rand_test.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+	testRandMu sync.Mutex
+	testRand   = rand.New(rand.NewPCG(0x5EED_1, 0x5EED_2))
+)
+
+// testRandFloat64 is a helper function to generate fixed sequence of random float64 values for testing.
+func testRandFloat64() float64 {
+	testRandMu.Lock()
+	defer testRandMu.Unlock()
+	return testRand.Float64()
+}
```
Comment (AlexanderYastrebov) [inline predicates/traffic/rand_test.go:18]: This is not reusable by multiple tests. Its better to introduce an object to generate random values from the same seed.

Comment (AlexanderYastrebov) [reply predicates/traffic/rand_test.go]: Fixed

**Comment by MustafaSaber** [comment]
👍

**Comment by AlexanderYastrebov** [comment]
:+1:

**Pull request merged**